blog/2025/01-01-efcore-and-me/index.mdx

```diff
@@ -0,0 +1,56 @@
+---
+title: EFCore refactoring is here
+description: EFCore, Jellyfin and me.
+authors: JPVenson
+slug: efcore-refactoring
+tags: [unstable, warning, efcore]
+---
+
+EFCore is here and this will have consequences.
+
+We finally reached our first milestone in cleaning up the legacy code regarding the database access, that means all SQL builders that were targeting SQLite directly are now removed from code.  
+
+This marks the first step towards a completely new Database design. But we now need to take a quick look ahead and see whats next.
+
+Please continue to read further to see what exactly that means and what the future brings.
+
+\- JPVenson
+
+{/* truncate */}
+
+## What exactly have we done
+
+[Josh’s blog post](../../2024/11-03-efcore-refactoring-incoming/index.mdx) in November 2024 outlined much of what we aimed to achieve, and I want to take this opportunity to address the feedback and questions we've received, clarifying a few key points.  
+
+The old code contained a lot of code that was directly accessing the underlying SQLite database. 
+SQLite is a file based database where Jellyfin is storing most of its important data.
+This has been an issue for 3 reasons.
+1. The old code was very poorly written
+2. The old code was very complex
+3. It was explicitly written to support SQLite only and nothing else  
+
+Because of this, we were not able to quickly extend our database it in a reliable way. The new code uses EntityFramework migrations which is the industry standard for database migrations.  
```

Comment:
Better clarity and removed a redundant it:
```suggestion
Because of this, we were unable to quickly and reliably extend our database. The new code uses EntityFramework migrations, which is the industry standard for database migrations.  
```